src/neuroconv/tools/spikeinterface/spikeinterface.py

```diff
@@ -570,6 +569,17 @@ def add_electrical_series(
     Missing keys in an element of metadata['Ecephys']['ElectrodeGroup'] will be auto-populated with defaults
     whenever possible.
     """
+    # TODO: remove completely after 10/1/2024
+    if compression is not None or compression_opts is not None:
+        warnings.warn(
+            message=(
+                "Specifying compression methods and their options at the level of tool functions has been deprecated. "
+                "Please use the `configure_backend` tool function for this purpose."
+            ),
+            category=DeprecationWarning,
+            stacklevel=2,
```

Comment:
Just to confirm, how are you thinking about the `stacklevel=2` here? I understand that this will raise the warning at the level of the function that calls this one (instead of raising it when you call this function). I guess that this makes sense because we want to let know users that use this function through the interface for example, right?

Reply:
Without it the printout doesn't mention what line actually triggered the warning which makes it harder to tell where and how to correct it; that's why we add `stacklevel=2` pretty much everywhere now

Reply:
And by the line that actually trigger it you mean the function that we are calling that ended up using the function where the warning is declared right?

Otherwise it just gives you the warning, right?

Reply:
I don't know why I am asking you this, sorry. I can test this myself quickly. Thanks for indulging me : P

Reply:
You got it. Nothing more annoying than having to ctrl+F on a GitHub repo looking for a warning string when I should be able to click the line in an IDE

Reply:
```python
import warnings

def function3(): 
    
    function2()

def function2():
    # This function calls function2
    function1()

def function1():
    # This will issue a warning with stacklevel=1
    warnings.warn("This is a stacklevel=1 warning", stacklevel=1)
    # This will issue a warning with stacklevel=2
    warnings.warn("This is a stacklevel=2 warning", stacklevel=2)

# Call function1, which in turn calls function2
function3()
```

Ouptut:
```
/tmp/ipykernel_260546/1160872622.py:13: UserWarning: This is a stacklevel=1 warning
  warnings.warn("This is a stacklevel=1 warning", stacklevel=1)
/tmp/ipykernel_260546/1160872622.py:9: UserWarning: This is a stacklevel=2 warning
  function1()
```